cryptography-providers/jdk/src/jvmMain/kotlin/algorithms/JdkEc.kt

```diff
@@ -133,24 +133,35 @@ internal sealed class JdkEc<PublicK : EC.PublicKey, PrivateK : EC.PrivateKey, KP
         private val key: JPublicKey,
     ) : EC.PublicKey, JdkEncodableKey<EC.PublicKey.Format>(key) {
         final override fun encodeToByteArrayBlocking(format: EC.PublicKey.Format): ByteArray = when (format) {
-            EC.PublicKey.Format.JWK -> error("$format is not supported")
-            EC.PublicKey.Format.RAW -> {
-                key as ECPublicKey
+            EC.PublicKey.Format.JWK            -> error("$format is not supported")
+            EC.PublicKey.Format.RAW            -> encodeToRaw(false)
+            EC.PublicKey.Format.RAW.Compressed -> encodeToRaw(true)
```

Comment:
It would be nice to use named parameters here:
```suggestion
            EC.PublicKey.Format.RAW            -> encodeToRaw(compressed = false)
            EC.PublicKey.Format.RAW.Compressed -> encodeToRaw(compressed = true)
```